bin/fetch-airlines.js

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env node
+
+/** eslint-disable node/shebang */
+
+// Airlines adaption
+
+const DIGITS_ANDOR_LETTERS = /^[0-9a-zA-Z]+$/;
+
+const OPENFLIGHTS_URL = "https://raw.githubusercontent.com/jpatokal/openflights/master/data/airlines.dat";
+const WAIT_BETWEEN_ATTEMPTS = 1e4;
+
+const fs = require("fs");
+const {resolve} = require("path");
+const {StringStream, DataStream} = require("scramjet");
+const {promisify} = require("util");
+
+const {get} = require("https");
+
+const awaitWritten = (stream, path) => {
+    return new Promise(
+        (s, j) => stream
+            .pipe(fs.createWriteStream(resolve(__dirname, path)))
+            .on("finish", s).on("error", j)
+    );
+};
+
+(async () => {
+    const fetch = () => new Promise(s => get(OPENFLIGHTS_URL, stream => s(stream)).end());
+
+    const columns = [
+        "airlineid", "name", "alias", "iata", "icao", "callsign", "country", "active"
+    ];
+    
+    let i = 0; let j = 0;
+    const stream = StringStream.from(async function() {
+        let retries = 5;
+        while (retries-- > 0) {
+            /** @type {import("http").IncomingMessage} */
+            const ret = await fetch();
+            if (ret.statusCode >= 200 && ret.statusCode < 300)
+                return ret;
+            if (ret.statusCode >= 400 && ret.statusCode < 500)
+                throw new Error("400 error!");
+            await new Promise(s => setTimeout(s, WAIT_BETWEEN_ATTEMPTS));
+
+            console.error("Retry...");
+        }
+        throw new Error("Too many errors...");
+    }, {})
+        .CSVParse()
+        .each(() => j++)
+        .map(x => x.map(col => col === "\\N" ? "" : col))
+        .map(x => {
+            const out = {};
+            for (let i = 0; i < columns.length; i++) {
+                if (i === 3 || i === 4) {
+                    // Ignore bad iatas and icaos.
+                    if(!x[i].match(DIGITS_ANDOR_LETTERS)) x[i] = "";
```

Comment:
Would this fix apply to airports also?

Reply:
Yes it would work with airports as well, if we change `if (i === 3 || i === 4)` to
`if (columns[i] == 'iata' || columns[I] == 'icao')`. Since i === 3 || 4 refers to the iata and icao columns, and those columns have different indexes in airports and airlines.